app/src/main/java/co/runloop/influencer/viewmodel/ContactsViewModel.java

```diff
@@ -9,13 +9,22 @@
 
 import co.runloop.influencer.data.ContactsProvider;
 import co.runloop.influencer.model.Contact;
+import co.runloop.influencer.model.Resource;
 import co.runloop.influencer.utils.DataObserver;
+import io.reactivex.Observable;
+import io.reactivex.ObservableOnSubscribe;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.schedulers.Schedulers;
 
 public class ContactsViewModel extends AndroidViewModel {
 
-    private MutableLiveData<List<Contact>> contacts;
+    private static final String TAG = ContactsViewModel.class.getSimpleName();
+
+    private MutableLiveData<Resource<List<Contact>>> contacts;
     private ContactsProvider provider;
     private DataObserver observer;
+    private CompositeDisposable disposables;
+    private boolean shouldRegisterObserver;
```

Comment:
Can you please explain how this is used? I am finding it a little confusing

Reply:
Disposable - object to stop chain of observables and release subscribers
Here just a list of disposables.
shouldRegisterObserver - we would like to be notified when contacts change, but if we have no permission, subscriber will be denied, and because we can lose our permission in any moment, we should set flag on SecureException to notify to setup our observer

Reply:
I understand the disposable object, I'm familiar with the RX patter. 

I was asking about the boolean flag. So, the observer is observing the contacts list or the permission?

Reply:
it observing contacts provider, but can be reseted by system when we lose permission, or just cannot be setup before we grant it

Reply:
contacts provider - i mean ContentResolver with contacts uri